src/colorprint.h

```diff
@@ -23,6 +24,10 @@ std::string FormatString(const char* msg, ...);
 void ColorPrintf(std::ostream& out, LogColor color, const char* fmt, va_list args);
 void ColorPrintf(std::ostream& out, LogColor color, const char* fmt, ...);
 
+// Returns true if the stream identified by 'f' appears to be a terminal that
+// supports colored output, false otherwise.
+bool IsColorTerminal(FILE* f);
```

Comment:
We only really support color output to `stdout`, not arbitrary files. Therefore I think this should be `StdoutIsColorTerminal()` or similar.